ntoskrnl/inbv/inbv.c

```diff
@@ -1066,13 +1092,21 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
              * Save the main image palette and replace it with black palette,
              * so that we can do fade-in effect later.
              */
-            BitmapInfoHeader = (PBITMAPINFOHEADER)Screen;
-            Palette = (LPRGBQUAD)((PUCHAR)Screen + BitmapInfoHeader->biSize);
+            BitmapInfoHeader = (PBITMAPINFOHEADER)BootLogo;
```

Comment:
Why not use BitBltWithoutPalette here too?

Reply:
we also save palette here

Reply:
Ah, right.